llvm/include/llvm/ProfileData/MemProf.h

```diff
@@ -289,10 +291,20 @@ struct IndexedAllocationInfo {
       : CallStack(CS.begin(), CS.end()), CSId(CSId), Info(MB) {}
 
   // Returns the size in bytes when this allocation info struct is serialized.
-  size_t serializedSize() const {
-    return sizeof(uint64_t) + // The number of frames to serialize.
-           sizeof(FrameId) * CallStack.size() +    // The callstack frame ids.
-           PortableMemInfoBlock::serializedSize(); // The size of the payload.
+  size_t serializedSize(IndexedVersion Version = Version0) const {
```

Comment:
For here and all other functions below that currently default this parameter to Version0: Since the current version is Version1, should that be the default? Or better - should Version be a required parameter?

Reply:
I was on the fence whether to use a required parameter or `= Version0`, but I chose to use `= Version0` to keep the patch small.  I'll switch to the required parameter.

Reply:
Or would it make sense to have a variable that tracks the current version and default to that?

Reply:
I have a slight preference to telling these functions (`serialize`, `serializedSize`, `deserialize`) exactly which version to use and leaving any smarts higher up.  This way, a very small number of places know what the current version is.